vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py

```diff
@@ -0,0 +1,155 @@
+
+        weight_symmetric = weight_quant.get("symmetric")
+        input_symmetric = input_quant.get("symmetric")
+
+        is_8_bits = weight_bit == input_bit == 8
+        is_tensor = weight_strategy == input_strategy == "tensor"
+        is_symmetric = weight_symmetric and input_symmetric
+
+        if is_8_bits and is_tensor and is_symmetric:
+            return CompressedTensorsW8A8StaticTensor(
+                fake_quant=self.fake_quant)
+        raise NotImplementedError(
+            "Scheme not supported. Only 8-bit static symmtetric "
+            "per tensor quantization is currently supported")
+
+    def get_scheme(self, layer: torch.nn.Module,
+                   layer_name: str) -> "CompressedTensorsScheme":
+        # TODO: How are layers which are combined in vllm listed in the ignore list?
+        if layer_name in self.ignore:
+            return CompressedTensorsUnquantized()
+
+        # TODO: need a better matching function; can adapt shared function with sparseml
+        layer_type_name = None
+        layer_name_class = type(layer).__name__.lower()
+        for target in self.layer_quant_details:
+            if target.lower() in layer_name_class:
+                layer_type_name = target
+                break
+
+        layer_quant_details = self.layer_quant_details.get(layer_type_name)
+        if layer_quant_details is None:
+            raise ValueError(
+                f"Could not find quantization details for {layer_name}.")
+        try:
+            return self._get_schema(weight_quant=layer_quant_details["weight"],
+                                    input_quant=layer_quant_details["input"])
+        except NotImplementedError as e:
+            raise e
+
+
+class CompressedTensorsLinearMethod(LinearMethodBase):
+
+    def __init__(self, quantization_config: CompressedTensorsConfig):
+        self.quantization_config = quantization_config
+
+    def create_weights(self, layer: torch.nn.Module, layer_name: str,
+                       input_size_per_partition: int,
+                       output_sizes_per_partition: List[int], input_size: int,
+                       output_size: int, params_dtype: torch.dtype,
+                       **extra_weight_attrs):
+        """
+        Use the CompressedTensorsScheme associated with each layer to create the 
+        necessary parameters for the layer.
+        """
+        weight_loader = extra_weight_attrs.get("weight_loader")
+
+        scheme = self.quantization_config.get_scheme(layer=layer,
+                                                     layer_name=layer_name)
+        scheme.create_weights(
+            layer=layer,
+            input_size_per_partition=input_size_per_partition,
+            output_sizes_per_partition=output_sizes_per_partition,
+            output_size=output_size,
+            params_dtype=params_dtype,
+            weight_loader=weight_loader)
+
+        set_weight_attrs(layer, {"scheme": scheme})
```

Comment:
we should set this directly as `layer` is not a weight its an `nn.Module`

`layer.scheme = scheme`